### PR TITLE
Viite 2101 check and update logic for templates

### PR DESCRIPTION
**Pull request by Campanha**

### CODE REVIEW

**Comment by Campanha** [comment]
all tests passing at http://livibuild04.vally.local/jenkins/job/viite/job/Run-Tests/135/console

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/JunctionDAO.scala [hunk 1]
```diff
@@ -167,21 +169,27 @@ class JunctionDAO extends BaseDAO {
...
-  def fetchTerminatedByRoadwayNumbers(roadwayNumbers: Iterable[Long]) : Seq[Junction] = {
-    val query =
-      s"""
+  def fetchHistoryByRoadwayNumbers(roadwayNumbers: Iterable[Long]) : Seq[Junction] = {
+    if (roadwayNumbers.nonEmpty) {
+      val query =
+        s"""
          SELECT j.ID, j.JUNCTION_NUMBER, j.NODE_NUMBER, j.START_DATE, j.END_DATE, j.VALID_FROM, j.VALID_TO, j.CREATED_BY, j.CREATED_TIME
          FROM JUNCTION j
-         LEFT JOIN JUNCTION_POINT jp ON j.ID = jp.JUNCTION_ID AND jp.VALID_TO IS NULL
```
Comment (skosonen) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/JunctionDAO.scala]: Why rows with valid_to set should be returned? If valid_to is set, the row is logically deleted.

Comment (skosonen) [reply digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/JunctionDAO.scala]: Okay, this is only used from the tests. Renaming this to fetchExpiredByRoadwayNumbers.

**Comment by Campanha** [comment]
all tests are now passing: http://livibuild04.vally.local/jenkins/job/viite/job/Run-Tests/148/console

**Pull request merged**